examples/lib/commons/circle_component.dart

```diff
@@ -18,4 +18,11 @@ class CircleComponent extends PositionComponent {
     super.render(canvas);
     canvas.drawCircle(Offset(radius, radius), radius, paint);
   }
+
+  @override
+  bool containsPoint(Vector2 point) {
+    final local = absoluteToLocal(point);
+    final center = Vector2.all(radius);
+    return local.distanceToSquared(center) <= radius * radius;
```

Comment:
I think we should convert this helper class to a `ShapeComponent` with a circle shape, since this is code duplication from what is in the `Circle` that extends `Shape`.

Reply:
Hmm, but the problem with the `Circle` is that it is not a `Component`; and it also keeps its own position and size, so can't be cleanly converted into a `PositionComponent` either.

Reply:
It can, you just use the `ShapeComponent`.

Reply:
Ah the shape keeps its initial position + the component position (and same for the other fields), how sloppy of me.
But that will be easy to fix, putting up a PR for that.

Reply:
You can keep this component like it is and I'll do the refactoring later when my PR is in. :)